client/src/components/ColorList.js

```diff
@@ -21,10 +23,36 @@ const ColorList = ({ colors, updateColors }) => {
     // Make a put request to save your updated color
     // think about where will you get the id from...
     // where is is saved right now?
+    if (isNaN(colorToEdit.id)) {
+      axiosWithAuth().post(`/api/colors`, colorToEdit)
+      .then(resp => {
+        // console.log(resp)
+        updateColors(resp.data)
+        setEditing(false)
+        // setColorToEdit(initialColor)
+      })
+      .catch(err => console.log(err.response))
+    } else {
+      axiosWithAuth().put(`/api/colors/${colorToEdit.id}`, colorToEdit)
+      .then(resp => {
+        // console.log(resp)
+        updateColors(colors.map(e => e.id === resp.data.id ? resp.data : e))
+        setEditing(false)
+        // setColorToEdit(initialColor)
+      })
+      .catch(err => console.log(err.response))
+    }
```

Comment:
Good usage of the `initialColor` object and the `isNaN` check to create a single dynamic form 💪